contracts/src/v0.8/functions/dev/1_0_0/FunctionsRouter.sol

```diff
@@ -0,0 +1,289 @@
+  struct Config {
+    // Flat fee (in Juels of LINK) that will be paid to the Router owner for operation of the network
+    uint96 adminFee;
+    // The function selector that is used when calling back to the Client contract
+    bytes4 handleOracleFulfillmentSelector;
+  }
+  Config private s_config;
+  event ConfigSet(uint96 adminFee, bytes4 handleOracleFulfillmentSelector);
+
+  error OnlyCallableByRoute();
+
+  // ================================================================
+  // |                       Initialization                         |
+  // ================================================================
+  constructor(
+    uint16 timelockBlocks,
+    uint16 maximumTimelockBlocks,
+    bool useAllowList,
+    address linkToken,
+    bytes32[] memory initialIds,
+    address[] memory initialAddresses,
+    bytes memory config
+  )
+    RouterBase(msg.sender, timelockBlocks, maximumTimelockBlocks, initialIds, initialAddresses, config)
+    AuthorizedOriginReceiver(useAllowList)
+    FunctionsSubscriptions(linkToken)
+  {}
+
+  /**
+   * @inheritdoc ITypeAndVersion
+   */
+  function typeAndVersion() public pure override returns (string memory) {
+    return "Functions Router v1";
+  }
+
+  // ================================================================
+  // |                 Configuration methods                        |
+  // ================================================================
+  /**
+   * @notice Sets the configuration for FunctionsRouter specific state
+   * @param config bytes of config data to set the following:
+   *  - adminFee: fee that will be paid to the Router owner for operating the network
+   */
+  function _setConfig(bytes memory config) internal override {
+    (uint96 adminFee, bytes4 handleOracleFulfillmentSelector) = abi.decode(config, (uint96, bytes4));
+    s_config = Config({adminFee: adminFee, handleOracleFulfillmentSelector: handleOracleFulfillmentSelector});
+    emit ConfigSet(adminFee, handleOracleFulfillmentSelector);
+  }
+
+  /**
+   * @inheritdoc IFunctionsRouter
+   */
+  function getAdminFee() external view override returns (uint96) {
+    return s_config.adminFee;
+  }
+
+  // ================================================================
+  // |                      Request methods                         |
+  // ================================================================
+
+  function _sendRequest(
+    bytes32 donId,
+    bool isProposed,
+    uint64 subscriptionId,
+    bytes memory data,
+    uint32 gasLimit
+  ) internal returns (bytes32) {
+    _isValidSubscription(subscriptionId);
+    _isValidConsumer(msg.sender, subscriptionId);
+
+    address route = this.getRoute(donId, isProposed);
+    IFunctionsCoordinator coordinator = IFunctionsCoordinator(route);
+
+    (, , address owner, , ) = this.getSubscription(subscriptionId);
+
+    (
+      bytes32 requestId,
+      uint96 estimatedCost,
+      uint256 gasAfterPaymentCalculation,
```

Comment:
Is there a better name for this?
Is this actually gasAfterPayment?

Reply:
Let's add a good comment here instead

Reply:
Did some renaming and added comments